core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/utils/ReplicatedTableProviderImpl.java

```diff
@@ -52,14 +61,66 @@ public boolean accept(String keyspace)
             return false;
         }
 
-        Set<TokenRange> tokenRanges = myMetadata.getTokenRanges(keyspace, myLocalhost);
+        KeyspaceMetadata keyspaceMetadata = myMetadata.getKeyspace(keyspace);
 
-        if (tokenRanges.isEmpty())
+        if (keyspaceMetadata != null)
         {
+            Map<String, String> replication = keyspaceMetadata.getReplication();
+            String replicationClass = replication.get(STRATEGY_CLASS);
+
+            switch(replicationClass)
+            {
+                case SIMPLE_STRATEGY:
+                    return validateSimpleStrategy(replication);
+                case NETWORK_TOPOLOGY_STRATEGY:
+                    return validateNetworkTopologyStrategy(keyspace, replication);
+                default:
+                    LOG.warn("Replication strategy of type {} is not supported", replicationClass);
+                    break;
+            }
+        }
+
+        return false;
+    }
+
+    private boolean validateSimpleStrategy(Map<String, String> replication)
+    {
+        int replicationFactor = Integer.parseInt(replication.get("replication_factor"));
```

Comment:
"replication_factor" should be declared as a constant just like STRATEGY_CLASS

Reply:
+1